Automattic/wp-calypso
# Reader Profile - Add user profile sub pages and use Posts API endpoint

Related to https://github.com/Automattic/loop/issues/266

## Proposed Changes

* This PR stubs out the additional Reader user profile sub-pages
* It implements the /users/$user_id/posts endpoint
* It uses TypeScript
* It creates our own controller underneath the /user-stream/ directory
* It's still behind the `reader/user-profile` flag.

Note that additional work needs to be done on styling. I'd like to follow up on that in future PRs.


<img width="1723" alt="Screenshot 2024-12-20 at 3 02 22 PM" src="https://github.com/user-attachments/assets/b887f194-605f-49b6-8ec4-19483b27007a" />



## Why are these changes being made?


* Push the Reader user profile project forward.

## Testing Instructions



* Using Calypso Live
* Go to /read/users/1351218 -- This path should not load because it's behind the feature flag
* Go to /read/users/1351218?flags=reader/user-profile -- The flag will load our /read/users/* paths
* View that the Posts load
* View that the tabs and their URL routes work.
* Try to break it
* Consider if it could leak outside of our feature flag somehow.

## Pre-merge Checklist
